src/main/java/org/opensearch/security/privileges/dlsfls/DlsFlsLegacyHeaders.java

```diff
@@ -0,0 +1,247 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+package org.opensearch.security.privileges.dlsfls;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.google.common.collect.Sets;
+
+import org.opensearch.Version;
+import org.opensearch.action.ActionRequest;
+import org.opensearch.action.admin.cluster.shards.ClusterSearchShardsRequest;
+import org.opensearch.cluster.metadata.Metadata;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.security.privileges.PrivilegesEvaluationContext;
+import org.opensearch.security.privileges.PrivilegesEvaluationException;
+import org.opensearch.security.support.Base64Helper;
+import org.opensearch.security.support.ConfigConstants;
+import org.opensearch.security.support.HeaderHelper;
+import org.opensearch.transport.Transport;
+import org.opensearch.transport.TransportRequest;
+
+/**
+ * Encapsulates functionality to provide transport headers with DLS/FLS information that need to be sent
+ * to nodes which run on the legacy DLS/FLS implementation. This is only needed for mixed clusters.
+ * See the attribute LEGACY_HEADERS_UNNECESSARY_AS_OF for the concrete version.
+ * <p>
+ * As soon as backward compat in mixed clusters is no longer required, this class should be removed.
+ *
+ */
+public class DlsFlsLegacyHeaders {
+    /**
+     * Defines the first OpenSearch version which does not need the legacy headers
+     * TODO this needs to be adapted
+     */
+    static final Version LEGACY_HEADERS_UNNECESSARY_AS_OF = Version.V_EMPTY;
```

Comment:
Important: Before this is releases, this must be set to the OpenSearch version this functionality is released in.